### PR TITLE
[VAL] Initial push for pytest engine

### PR DESCRIPTION
**Pull request by MateuszGrabuszynski**
Added as a PR to allow reviews. This PR should allow further development of pytest-based validation scenarios.

* Moved the validation apps into `.../tests/validation/application`
* Aligned docs to accommodate that change (`..` added in README.md)
* Added the necessary pytest engine files from internal repo (minimal viable set of them), files were reviewed briefly, headers aligned to BSD license with original authors' permission

### CODE REVIEW

File: tests/validation/pytest/Engine/fixtures.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        media = "/mnt/media"
+    os.environ["media"] = media
+    return media
+
+
+@pytest.fixture(scope="session")
+def build(request):
+    build = request.config.getoption("--build")
+    if build is None:
+        build = "../Media-Transport-Library"
```
[the remark this answers is not in the document]
Comment (MateuszGrabuszynski) [reply tests/validation/pytest/Engine/fixtures.py]: Fixed

File: tests/validation/pytest/Engine/application_config.py
[the remark this answers is not in the document]
Comment (MateuszGrabuszynski) [reply tests/validation/pytest/Engine/application_config.py:1]: Removed

File: tests/validation/pytest/Engine/ffmpeg_app.py
[the remark this answers is not in the document]
Comment (MateuszGrabuszynski) [reply tests/validation/pytest/Engine/ffmpeg_app.py:1]: Removed

**Review by KarolinaPomian — COMMENTED** [review]
Readme for pytest would be usefull.

**Comment by MateuszGrabuszynski** [comment]
> Readme for pytest would be usefull.

Great remark. Will start adding it at some point where the implementation is more stable (not sure when at the moment) :)

**Pull request merged**